codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -239,6 +239,7 @@ void removeStream(DefaultStream stream) {
         private State state = IDLE;
         private short weight = DEFAULT_PRIORITY_WEIGHT;
         private DefaultStream parent;
+        // TODO(nathanmittler): We no longer need a map. LinkedList?
```

Comment:
We currently don't expose it externally but we do rely upon it internally (see takeChild method [put](https://github.com/netty/netty/blob/master/codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java#L576) and [remove](https://github.com/netty/netty/blob/master/codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java#L581) also removeChild [remove](https://github.com/netty/netty/blob/master/codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java#L596)).  I'm not sure if the "put" is necessary to be a map operation....especially after a PR I have pending is submitted but the "remove" seems like a map operation....WDYT?